test/test_integration.rb

```diff
@@ -69,16 +69,8 @@ def wait_booted
   def restart_server(connection)
     signal :USR2
 
-    sleep 1
-
     connection.write "GET / HTTP/1.1\r\n\r\n" # trigger it to start by sending a new request
 
-    assert_raises Errno::ECONNRESET do
```

Comment:
i guess the restarting is already tested by the wait_for_server_to_boot, because it looks for "Use Ctrl-C to stop" which won't appear if the server didn't restart.